platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/roster/RosterRetriever.java

```diff
@@ -241,12 +242,15 @@ public static Roster buildRoster(@Nullable final AddressBook addressBook) {
         if (addressBook == null) {
             return null;
         }
+
+        final List<RosterEntry> entries = addressBook.getNodeIdSet().stream()
+                .map(addressBook::getAddress)
+                .map(RosterRetriever::buildRosterEntry)
+                .sorted(Comparator.comparing(RosterEntry::nodeId))
+                .toList();
+
         return Roster.newBuilder()
-                .rosterEntries(addressBook.getNodeIdSet().stream()
-                        .map(addressBook::getAddress)
-                        .map(RosterRetriever::buildRosterEntry)
-                        .sorted(Comparator.comparing(RosterEntry::nodeId))
-                        .toList())
+                .rosterEntries(new ArrayList<>(entries)) // create a new list that is mutable for testing
```

Comment:
I'm unsure if this is a good idea.

Generally, mutable collections are dangerous and should be avoided, especially in production code. And the RosterRetriever is a production code for the most part.

Specifically to the Roster, this data structure is considered immutable. Trying to create "mutable" instances leads to testing something that we've never designed or intended. And certainly, something that we (should) never use in production code. So from this perspective, the value of such tests is unclear.

I suggest to find a way to avoid creating mutable instances of this class.

Reply:
I generally agree. The reason for making these mutable was to mimic how the address book supported adding and removing addresses from the same instance - and this was used a fair bit in testing. If the intention is to never mutate the roster, then I can remove this and re-write the tests to not do this either.

Reply:
Rosters must be immutable once built since their hash value is used as a reference to them.

Reply:
Got it! I've updated the PR to remove the mutable parts. Please re-review when you two get the chance.